torch/csrc/jit/script/compiler.cpp

```diff
@@ -2241,8 +2241,39 @@ struct to_ir {
         auto values = getValues(ll.inputs(), /*maybe_unpack=*/true);
         return graph->insertNode(graph->createTuple(values))->output();
       } break;
+      case TK_DICT_LITERAL: {
+        auto dl = DictLiteral(tree);
+        auto keys = getValues(dl.key_inputs(), /*maybe_unpack=*/true);
+        auto values = getValues(dl.value_inputs(), /*maybe_unpack=*/true);
+
+        JIT_ASSERT(keys.size() == values.size());
+
+        TypePtr key_type = nullptr;
+        TypePtr value_type = nullptr;
+
+        if (type_hint && type_hint->kind() == TypeKind::DictType) {
+          auto dict_type = type_hint->expect<DictType>();
+          key_type = dict_type->getKeyType();
+          value_type = dict_type->getValueType();
+        } else if (!values.empty()) {
+          key_type = keys.at(0)->type();
+          value_type = values.at(0)->type();
+        } else {
+          throw ErrorReport(tree) << "Cannot get key and value types without a"
```

Comment:
For lists, we just assume tensor. For dict, we could consider assuming string -> tensor. I am not really sure how people will typically use dicts, however.